server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -61,8 +61,6 @@
    * This method may trigger a {@link #persistAll(Committer)} using the supplied Committer. If it does this, the
    * Committer is guaranteed to be *created* synchronously with the call to add, but will actually be used
    * asynchronously.
-   * <p>
-   * The add, clear, persist, persistAll, and push methods should all be called from the same thread.
```

Comment:
Btw, the original idea was that the methods that accept Committer objects should all be called from the same thread, because that ensures that the committed metadata on disk is in sync with the data itself.